scripts/ci/generate_install_lists.py

```diff
@@ -74,6 +74,11 @@ def main(argv=sys.argv[1:]):
                 package_root = all_packages[package]
                 Path(package_root, 'COLCON_IGNORE').touch()
 
+        # generate a list of non-ROS package directories
+        non_ros_package_paths = set(
+            d for d in selected_packages.values()
+            if not os.path.isfile(os.path.join(d, 'package.xml')))
```

Comment:
Not sure if there is a better way of detecting ROS packages, but we have some packages not-really-ros-packages shipping convenient package.xml like DART https://github.com/dartsim/dart/blob/main/package.xml. Do we want (or it is fine) to consider them ROS packages here?

Reply:
It isn't really a question of what actually is or isn't a ROS package, but rather what rosdep thinks is a ROS package.

The goal is to make sure that rosdep never crawls deeper than any directory that colcon has identified as a package root. If there is a `package.xml` in that directory, then we know `rosdep` will stop crawling on its own. If there isn't, then rosdep has no business in there to begin with, and it should be specifically excluded from the crawl.

There is a better way to detect actual ROS packages, though. We could get the package types from the call to `colcon list` look for packages which have a type that starts with `ros.` (i.e. `ros.ament_cmake`, `ros.ament_python`, `ros.cmake`, etc). This *should* correspond to packages that contain a `package.xml`, it's just a lot easier to check for the manifest directly.

Reply:
> Not sure if there is a better way of detecting ROS packages, but we have some packages not-really-ros-packages shipping convenient package.xml like DART

As Scott mentioned, what we really want to do is skip packages which _don't_ have a package.xml for rosdep to inspect and so any package.xmls of convenience should be perfectly usable and we would want those packages to remain unskipped here.